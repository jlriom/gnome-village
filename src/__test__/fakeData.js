export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
