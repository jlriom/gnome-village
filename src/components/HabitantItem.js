import React from "react";
import Media from "react-bootstrap/Media";
import Image from "react-bootstrap/Image";
import LazyLoad from "react-lazyload";

const HabitantItem = habitant => {
	const {
		id,
		name,
		thumbnail,
		age,
		weight,
		height,
		hair_color,
		professions,
		friends
	} = habitant.habitant;
	return (
		<>
			<span className='d-inline-block m-4' key={id}>
				<Media>
					<LazyLoad once>
						<Image
							className='d-inline-block mx-4'
							src={thumbnail}
							alt={name}
							rounded
							style={{ height: "10rem", width: "10rem" }}
						/>
					</LazyLoad>
					<Media.Body style={{ width: "100%" }}>
						<h5>{name}</h5>
						<hr />
						<div className='container'>
							<small>
								<div className='row'>
									<div className='col-lg-4'>
										<dl className='row'>
											<dt className='col-sm-5'>Age:</dt>
											<dd className='col-sm-7'> {age}</dd>
											<dt className='col-sm-5'>Hair Color:</dt>
											<dd className='col-sm-7'> {hair_color}</dd>
											<dt className='col-sm-5'>Weight:</dt>
											<dd className='col-sm-7'> {weight.toFixed(3)}</dd>
											<dt className='col-sm-5'>Height:</dt>
											<dd className='col-sm-7'> {height.toFixed(3)}</dd>
										</dl>
									</div>
									<div className='col-lg-4'>
										{professions && professions.length > 0 ? (
											<>
												<b>Professions </b> <br />
												{professions.map((profession, index, professions) => (
													<span key={profession}>
														{profession}
														{professions.length - 1 > index ? ", " : ""}
													</span>
												))}
											</>
										) : (
											<span></span>
										)}
									</div>
									<div className='col-lg-4'>
										{friends && friends.length > 0 ? (
											<>
												<b>Friends </b> <br />
												{friends.map((friend, index, friends) => (
													<span key={friend}>
														{friend}
														{friends.length - 1 > index ? ", " : ""}
													</span>
												))}
											</>
										) : (
											<span></span>
										)}
									</div>
								</div>
							</small>
						</div>
					</Media.Body>
				</Media>
			</span>
		</>
	);
};

export default HabitantItem;
