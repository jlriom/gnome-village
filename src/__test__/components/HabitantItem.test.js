import React from "react";
import HabitantItem from "../../components/HabitantItem";
import renderer from "react-test-renderer";
import { habitant } from "../fakeData";
import { MemoryRouter } from 'react-router-dom';

const props = {
    habitant: habitant,
    manageFriends: false,
    myId: 0,
    myfriends: [],
    addFriend: jest.fn(),
    removeFriend: jest.fn(),
    addRemoveFriend: jest.fn()
};

describe("<HabitantItem />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <HabitantItem {...props} />
            </MemoryRouter>
        );
        expect(tree).toMatchSnapshot();
    });
});
