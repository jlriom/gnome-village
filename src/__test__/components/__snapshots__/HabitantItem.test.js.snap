// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitantItem /> renders ok 1`] = `
<span
  className="d-inline-block m-4 full-width"
>
  <div
    className="full-width media"
  >
    <div
      className="lazyload-placeholder"
      style={
        Object {
          "height": undefined,
        }
      }
    />
    <div
      className="full-width media-body"
    >
      <h5
        className="d-flex justify-content-between"
      >
        <span>
          Cogwitz Bittergrill
        </span>
         
      </h5>
      <hr />
      <div
        className="container"
      >
        <small>
          <div
            className="row"
          >
            <div
              className="col-lg-4"
            >
              <dl
                className="row full-width"
              >
                <dt
                  className="col-sm-6"
                >
                   
                  Age:
                </dt>
                <dd
                  className="col-sm-6"
                >
                   
                  127
                </dd>
                <dt
                  className="col-sm-6"
                >
                   
                  Hair Color:
                </dt>
                <dd
                  className="col-sm-6"
                >
                   
                  Black
                </dd>
                <dt
                  className="col-sm-6"
                >
                   
                  Weight:
                </dt>
                <dd
                  className="col-sm-6"
                >
                   
                  40.075
                </dd>
                <dt
                  className="col-sm-6"
                >
                   
                  Height:
                </dt>
                <dd
                  className="col-sm-6"
                >
                   
                  93.663
                </dd>
              </dl>
            </div>
            <div
              className="col-lg-4"
            >
              <div
                className="row full-width"
              >
                <b>
                  Professions: 
                </b>
                 
                <span>
                  Prospector
                  , 
                </span>
                <span>
                   Tinker
                  , 
                </span>
                <span>
                  Baker
                  , 
                </span>
                <span>
                  Tailor
                  , 
                </span>
                <span>
                  Leatherworker
                  , 
                </span>
                <span>
                  Sculptor
                  
                </span>
              </div>
            </div>
            <div
              className="col-lg-4"
            >
              <div
                className="row full-width"
              >
                <b>
                  Friends: 
                </b>
                  
                <span>
                  <a
                    href="/brastlewark/Whitwright Fussbuster"
                    onClick={[Function]}
                  >
                    Whitwright Fussbuster
                  </a>
                  , 
                </span>
                <span>
                  <a
                    href="/brastlewark/Fizwood Bitterrocket"
                    onClick={[Function]}
                  >
                    Fizwood Bitterrocket
                  </a>
                  , 
                </span>
                <span>
                  <a
                    href="/brastlewark/Fizwood Felpower"
                    onClick={[Function]}
                  >
                    Fizwood Felpower
                  </a>
                  
                </span>
              </div>
            </div>
          </div>
        </small>
      </div>
    </div>
  </div>
</span>
`;
